### PR TITLE
Mark existing `multi_ptr` tests as legacy

### PR DESCRIPTION
**Pull request by yuriykoch**
Make existing `multi_ptr` tests legacy according to the SYCL-2020 updates:

- enforce legacy decoration usage
- rename existing files and test names
- add SYCL_CTS_ENABLE_LEGACY_TESTS flag
- and update the run_conformance_tests.py appropriately, to make it possible to disable legacy tests

### CODE REVIEW

**Comment by psalz** [comment]
In #217 I'm referring to tests that use the current (soon to be old) way of writing tests as "legacy". To avoid confusion, should we simply call these something like "1.2.1-tests" instead?

**Comment by psalz** [comment]
After looking some more into this; I originally thought you were guarding tests for SYCL 1.2.1 functionality that has been removed/made optional in 2020. However it is my understanding that `access::decorated::legacy` is a non-optional part of the 2020 specification, i.e. it is required for conformance, so why should there be an option for the CTS to disable testing for this?

**Comment by yuriykoch** [comment]
> However it is my understanding that `access::decorated::legacy` is a non-optional part of the 2020 specification, i.e. it is required for conformance, so why should there be an option for the CTS to disable testing for this?

This indeed is a non-optional, but deprecated part of the SYCL-2020 standard. Therefore we might use it, for example, for deprecated accessor API. The option is enabled by default, so it doesn't affect the conformance. Still it provides a possibility to have tests for deprecated features disabled - and to state explicitly the tests are on the deprecated feature which may be gone someday.
May be we need to rename the `legacy` option/macro to the `deprecated` within this PR?

**Review by bader — APPROVED** [review]

File: run_conformance_tests.py [hunk 1]
```diff
@@ -59,6 +59,11 @@ def handle_args(argv):
         help='Disable full conformance mode to avoid extensive tests.',
         required=False,
         action='store_true')
+    parser.add_argument(
+        '--enable-legacy',
+        help='Enable legacy tests.',
+        required=False,
+        action='store_true')
```
Comment (psalz) [inline run_conformance_tests.py:66]: Shouldn't this also default to true for the conformance report generation?

**Review by psalz — APPROVED** [review]
Sorry for the late response. I agree that having a switch for disabling tests for deprecated features could be useful, as long as they are enabled by default. I'd still be in favor of renaming the option to something like `SYCL_CTS_TEST_DEPRECATED_FEATURES` (or `SYCL_CTS_ENABLE_DEPRECATED_TESTS`, although that sounds like the tests themselves are deprecated, which they are not) to avoid confusion with "legacy-style tests" once the Catch2 migration has been merged.

File: run_conformance_tests.py (hunk 1 above)
Comment (yuriykoch) [reply run_conformance_tests.py:66]: My bad. Fixed both the default value and argument naming by 97f90a65ce88610ee4af1a99d5ff183812b26ae0.

**Comment by yuriykoch** [comment]
Thanks, the suggested `SYCL_CTS_TEST_DEPRECATED_FEATURES` looks good to me. Updated by bf9cdf6ba43771fb6802354cae2eacc6e37e554e

**Review by psalz — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**